src/Aspire.Hosting/Orchestrator/ApplicationOrchestrator.cs

```diff
@@ -183,6 +183,11 @@ public async Task StopAsync(CancellationToken cancellationToken)
         await _dcpExecutor.StopAsync(cancellationToken).ConfigureAwait(false);
     }
 
+    public async ValueTask DisposeAsync()
+    {
+        await this.StopAsync(CancellationToken.None).ConfigureAwait(false);
```

Comment:
ApplicationOrchestrator wraps DcpExecutor, so disposing the ApplicationOrchestrator automatically stops the DcpExecutor. Do they both need to be disposable? The DI container will end up disposing both of them (and them the orchestrator will also dispose the executor).

There isn't a huge negative impact other than executor being disposed twice, but it looks like you already have code to make that safe.

Reply:
That is a great point, James. I did not realize the DcpExecutor was part of the component set. Once I changed it to be IAsyncDisposable, there is no need to change the ApplicationOrchestrator. Fixed.